client/lib/upgrades/actions/domain-management.js

```diff
@@ -126,17 +125,17 @@ function fetchDomains( siteId ) {
 		} );
 	}
 
+	const domains = DomainsStore.getBySite( siteId );
+	if ( domains.isFetching || domains.hasLoadedFromServer ) {
```

Comment:
@aidvu: The `hasLoadedFromServer` check exists in the other functions because they're called directly from `componentWillUpdate` in the data components. If there was no check within the action creator, the fetching would keep happening over and over. With the `DomainsStore` it's different because the data components have a [special check](https://github.com/Automattic/wp-calypso/blob/0d83a6d23bce22165520146e54a59ef04b63909c/client/components/data/domain-management/nameservers/index.jsx#L54) to only call `fetchDomains` when the site changes.

For the future I feel we should keep as much logic out of the data components as possible (they are really an anti-pattern) and instead move the check into the `fetchDomains` call itself. That said, if you'd like to move forward without additional refactoring, you can skip the `hasLoadedFromServer` check here.

@stephanethomas:

> What if we want to force a refresh of the list of domains?

Maybe it would be better to make that explicit with a separate `refreshDomains` action creator. We're going to keep getting confused if we use `fetchDomains` for both a) initially loading the data and b) refetching the data later on.